library/Fission/AWS/Route53.hs

```diff
@@ -43,12 +46,53 @@ registerDomain username (CID hash) = do
 
   changeRecord Cname baseUrl (getGateway gateway) >>= \case
     Left err -> return <| Left err
-    Right _ -> 
+    Right _ ->
       changeRecord Txt dnslinkUrl (dnslink `wrapIn` "\"") >>= \case
         Left err -> return <| Left err
         Right _ -> return <| Right <| AWS.DomainName baseUrl
 
 changeRecord ::
+  ( MonadRIO            cfg m
+  , MonadUnliftIO           m
+  , HasLogFunc          cfg
+  , Has AWS.AccessKey   cfg
+  , Has AWS.SecretKey   cfg
+  , Has AWS.ZoneID      cfg
+  , Has AWS.MockEnabled cfg
+  )
+  => RecordType
+  -> Text
+  -> Text
+  -> m (Either ServerError ChangeResourceRecordSetsResponse)
+changeRecord recordType domain content = do
+  AWS.MockEnabled mockAws <- Config.get
+  if mockAws
+    then changeRecordMock recordType domain content
+    else changeRecord' recordType domain content
+
+-- | Mock changing the given DNS record on Route53
+changeRecordMock :: (MonadRIO cfg m, HasLogFunc cfg)
+  => RecordType
+  -> Text
+  -> Text
+  -> m (Either ServerError ChangeResourceRecordSetsResponse)
+changeRecordMock recordType domain content = do
+    logDebug mockMessage
+    return (Right mockRecordResponse)
+  where
+    mockMessage = "MOCK: Updating DNS "
+                  <> displayShow recordType
+                  <> " record at: "
+                  <> displayShow domain
+                  <> " with "
+                  <> displayShow content
+    mockTime = parseISODateTime "2011-11-19 18:28:52.607875 UTC"
```

Comment:
anything in particular about this time that we need in the mock?

Otherwise, we could use `getCurrentTime` from http://hackage.haskell.org/package/time-1.6.0.1/docs/Data-Time-Clock.html and save the import